### PR TITLE
[AIRFLOW-3783] Switch to HEADER option being provided already by Redshift

### PR DESCRIPTION
**Pull request by shrechak**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3783

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

    Redshift has added a new HEADER option to the UNLOAD operation parameters.So we can remove the default PARALLEL OFF in the existing RedshiftToS3Transfer operator. The unload operation can now run parallely, hence faster, along with headers, if required.

### Tests

- [x] My PR updates the following unit tests:
     TestRedshiftToS3Transfer

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by kurtqq** [comment]
is this goes in next release?

**Comment by Fokko** [comment]
@kurtqq No, the `v1.10.3` is in RC2, and therefore we won't add any new functionality. You could speed up the process by giving the operator a try.

@shrechak Could you rebase?

**Comment by ashb** [comment]
Does this need a certain version of Redshift cluster? i.e. will this still work on clusters that have been around for some time?

**Comment by kurtqq** [comment]
@ashb  to answer your question - yes.
"Versions 1.0.3945, 1.0.4081, 1.0.4222
Time period: September 19–October 10, 2018
Features and Improvements
    Amazon Redshift: You can specify the HEADER option in the UNLOAD command to add a specific header row to each created output file. The header row contains the column names created by the unload query.
"
https://docs.aws.amazon.com/redshift/latest/mgmt/rs-mgmt-cluster-version-notes.html

also info on stackoverflow (see fez answer):
https://stackoverflow.com/questions/24681214/unloading-from-redshift-to-s3-with-headers

**Comment by ashb** [comment]
In which case this probably needs to be an option (that can default to on) to continue working with older clusters, right? Or do clusters auto/forcibly get upgraded?

**Comment by kurtqq** [comment]
@ashb It's auto/forcibly update:

`Amazon Redshift provides a setting, Allow Version Upgrade, to specify whether to automatically upgrade the Amazon Redshift engine in your cluster if a new version of the engine becomes available. This setting doesn't affect the database version upgrades, which are applied during the maintenance window that you specify for your cluster. Amazon Redshift engine upgrades are major version upgrades, and Amazon Redshift database upgrades are minor version upgrades. You can disable automatic version upgrades for major versions only. For more information about maintenance windows for minor version upgrades, see Maintenance Windows. `
https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-clusters.html

The 1.0 is the cluster version which the user can control if he wants to update or not. 
but the user have no control over the database version update. Redshift force update this in the maintenance windows. 
Since all users run the same cluster version (1.0)  this PR is safe to be merged in my opinion.

**Review by ashb — APPROVED** [review]
Not familiar with RedShift but the logic seems sensible.

**Comment by OmerJog** [comment]
@shrechak  Can you please rebase ?

**Comment by OmerJog** [comment]
@shrechak you have minor style errors to fix

flake8

./tests/operators/test_redshift_to_s3_operator.py:50:20: E999 TabError: inconsistent use of tabs and spaces in indentation





mypy

tests/operators/test_redshift_to_s3_operator.py:50: error: inconsistent use of tabs and spaces in indentation

File: tests/operators/test_redshift_to_s3_operator.py [hunk 1]
```diff
@@ -45,7 +45,16 @@ def test_execute(self, mock_get_conn, mock_run, mock_Session):
         table = "table"
         s3_bucket = "bucket"
         s3_key = "key"
-        unload_options = ('PARALLEL OFF',)
+        unload_options = ('HEADER',)
+
+	columns_query = """
```
Comment (ashb) [inline tests/operators/test_redshift_to_s3_operator.py:50]:
```suggestion
        columns_query = """
```

**Review by ashb — CHANGES_REQUESTED** [review]
Fix the style issues.

**Comment by OmerJog** [comment]
@shrechak if you don't mind fixing the minor styles maybe this PR will make it to the next release

**Comment by OmerJog** [comment]
@shrechak ping

**Comment by feluelle** [comment]
Closed in favor of #6309

**Pull request closed without merging**